ui/src/lib/utils/http.ts

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2024-Present The UDS Authors
+import { apiAuthEnabled } from '$lib/features/api-auth/store'
+import { fetchConfig } from '$lib/utils/helpers'
+import { get } from 'svelte/store'
+
+const BASE_URL = '/api/v1'
+
+const headers = new Headers({
+  'Content-Type': 'application/json',
+})
+
+export class HTTP {
+  constructor() {
+    const token = sessionStorage.getItem('token') || ''
+    const isApiAuthEnabled = get(apiAuthEnabled)
+    if (!token && isApiAuthEnabled) {
+      this.invalidateAuth()
+    }
+  }
+
+  // Updates the internal token used for authentication.
+  updateToken(token: string) {
+    sessionStorage.setItem('token', token)
+  }
+
+  private invalidateAuth() {
```

Comment:
i think you can use the `#` syntax here, like in the main store.ts.

```suggestion
  #invalidateAuth() {
```

Reply:
whoops, didnt like that one. Need it to be private, otherwise not able to call it in the constructor above.

Reply:
sorry if you already tried this, but in the constructor will `this.#invalidateAuth()` work?

Reply:
ah nice, yeah it liked that